crates/primitives/starknet/src/sequencer_address/mod.rs

```diff
@@ -0,0 +1,114 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use scale_codec::{Decode, Encode};
+use sp_inherents::{InherentData, InherentIdentifier, IsFatalError};
+use thiserror_no_std::Error;
+
+/// The identifier for the `sequencer_address` inherent.
+pub const INHERENT_IDENTIFIER: InherentIdentifier = *b"seqaddr0";
+
+/// Default value in case the sequencer address is not set.
+pub const DEFAULT_SEQUENCER_ADDRESS: [u8; 32] =
+    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 222, 173];
+
+/// The storage key for the sequencer address value.
+pub const SEQ_ADDR_STORAGE_KEY: &[u8] = b"starknet::seq_addr";
+
+/// The inherent type for the sequencer address.
+pub type InherentType = [u8; 32];
+
+#[derive(Encode, sp_runtime::RuntimeDebug)]
+#[cfg_attr(feature = "std", derive(Decode, Error))]
```

Comment:
Because you are using `thiserror_no_std` you can derive Error even when not in `std`.

Event if it is not used, it's always good to have it implemented for an error type.

Same with `Decode`. Event if only `Encode` is used right now, it is not absurd to also derive `Decode`